### ISSUE TITLE
Unit equivalency in Python

### ISSUE THREAD

**Issue by Igrom** [hgrecco/pint#455]
Issue when running code in a previous version of Python. Python 2.7, Pint 0.7.2:
```
import pint
ureg = pint.UnitRegistry()

width = ureg.Quantity('100mm')
height = ureg.Quantity('150mm')

print(repr(width.u)) //<Unit('millimeter')>
print(repr(height.u)) //<Unit('millimeter')>

assert width.u == height.u, 'The units are equal.'
assert width.u != height.u, 'The units are different.'
```

Both assertion statements pass. Meanwhile, Python 3.5.2 correctly throws an error.

**Comment by onealj** [comment]
This looks like this was fixed after the release of 0.7.2 by defining
_Unit.__ne__(self, other)

https://github.com/hgrecco/pint/commit/96d82ab86e65ccc7c69b5ab2b6f42d45e3ce5973

**Comment by onealj** [comment]
This should be closed as a duplicate of #372

**Comment by hgrecco** [comment]
Indeed!
